parsec/api/data/base.py

```diff
@@ -70,13 +61,17 @@ def __new__(cls, name, bases, nmspc):
             nmspc["SCHEMA_CLS"], DataValidationError, DataSerializationError
         )
 
-        return cls.CLS_ATTR_COOKING(raw_cls)
+        return raw_cls
 
 
 class SignedDataMeta(DataMeta):
     BASE_SCHEMA_CLS = BaseSignedDataSchema
 
 
+T = TypeVar("T", bound="BaseSignedData")
```

Comment:
name this `BaseSignedDataType` like we have `BaseDataType` ? (using `T` like this give Vietnam flashback to the old C++ dev I am...)

Also maybe the we can call that `ConcreteData`/`ConcreteSignedData` to oppose to the base prefix ? (the `*Type` suffix feel like we are doing metaclass stuff)

Reply:
Is `Base` still a problem as TypeVar class are named like `BaseDataTypeVar` . As `BaseData` and `BaseSignedData` are never directly instantiated, keeping `Base` in the name sounds to be a good idea to me.